src/neptune_scale/sync/sync_process.py

```diff
@@ -595,47 +607,60 @@ def _is_fatal_error(ex: Exception) -> bool:
 
 class FileUploaderThread(Daemon):
     def __init__(
-        self, project: str, api_token: str, operations_repository: OperationsRepository, errors_queue: ErrorsQueue
+        self,
+        project: str,
+        api_token: str,
+        operations_repository: OperationsRepository,
+        errors_queue: ErrorsQueue,
+        max_concurrent_uploads: int = MAX_CONCURRENT_FILE_UPLOADS,
     ) -> None:
         super().__init__(name="FileUploaderThread", sleep_time=1)
 
         self._project = project
         self._neptune_api_token = api_token
         self._operations_repository = operations_repository
         self._errors_queue = errors_queue
+        self._max_concurrent_uploads = max_concurrent_uploads
 
         self._api_client: Optional[ApiClient] = None
 
-    def close(self) -> None:
-        if self._api_client is not None:
-            self._api_client.close()
+        # AIO loop used for file uploads. When created (lazily), it is set as this thread's event loop.
+        self._aio_loop: Optional[asyncio.AbstractEventLoop] = None
+
+        # Shared transport and session to take advantage of connection pooling.
+        self._aio_session: Optional[requests.Session] = None
+        self._aio_transport: Optional[AsyncioRequestsTransport] = None
 
     def work(self) -> None:
         try:
-            while file_upload_requests := self._operations_repository.get_file_upload_requests(10):
+            while file_upload_requests := self._operations_repository.get_file_upload_requests(
+                self._max_concurrent_uploads
+            ):
                 logger.debug(f"Have {len(file_upload_requests)} file upload requests to process")
 
+                # No networking was done yet - initialize lazily
                 if not self._api_client:
                     self._api_client = backend_factory(self._neptune_api_token)
 
+                    self._aio_loop = asyncio.new_event_loop()
+                    asyncio.set_event_loop(self._aio_loop)
+
+                    self._aio_session = requests.Session()
```

Comment:
Why is this a field in the class? Can't the transport initialize its own session?